example/probe-demo/probe-demo.cpp

```diff
@@ -0,0 +1,247 @@
+    "    all_i_ion   total ionic membrane current [nA] in each CV\n"
+    "    all_i_na    sodium ion membrane current [nA] in each CV\n"
+    "    all_i_k     potassium ion membrane current [nA] in each CV\n"
+    "    all_i       total membrane current [nA] in each CV\n"
+    "    all_c_na    internal sodium concentration [mmol/L] in each CV\n"
+    "    all_c_k     internal potassium concentration [mmol/L] in each CV\n"
+    "    all_hh_m    HH state variable m in each CV\n"
+    "    all_hh_h    HH state variable h in each CV\n"
+    "    all_hh_n    HH state variable n in each CV\n";
+
+struct options {
+    double sim_end = 100.0;   // [ms]
+    double sim_dt = 0.025;    // [ms]
+    double sample_dt = 1.0;   // [ms]
+    unsigned n_cv = 10;
+
+    bool scalar_probe = true;
+    any probe_addr;
+    std::string value_name;
+};
+
+const char* argv0 = "";
+bool parse_options(options&, int& argc, char** argv);
+void vector_sampler(arb::cell_member_type, arb::probe_tag, any_ptr, std::size_t, const arb::sample_record*);
+void scalar_sampler(arb::cell_member_type, arb::probe_tag, any_ptr, std::size_t, const arb::sample_record*);
+
+struct cable_recipe: public arb::recipe {
+    arb::cable_cell_global_properties gprop;
+    any probe_addr;
+
+    explicit cable_recipe(any probe_addr, unsigned n_cv):
+        probe_addr(std::move(probe_addr))
+    {
+        gprop.default_parameters = arb::neuron_parameter_defaults;
+        gprop.default_parameters.discretization = arb::cv_policy_fixed_per_branch(n_cv);
+    }
+
+    arb::cell_size_type num_cells() const override { return 1; }
+    arb::cell_size_type num_probes(arb::cell_gid_type) const override { return 1; }
+    arb::cell_size_type num_targets(arb::cell_gid_type) const override { return 0; }
+
+    arb::probe_info get_probe(arb::cell_member_type probe_id) const override {
+        return {probe_id, 0, probe_addr}; // (tag value 0 is not used)
+    }
+
+    arb::cell_kind get_cell_kind(arb::cell_gid_type) const override {
+        return arb::cell_kind::cable;
+    }
+
+    arb::util::any get_global_properties(arb::cell_kind) const override {
+        return gprop;
+    }
+
+    arb::util::unique_any get_cell_description(arb::cell_gid_type) const override {
+        using namespace arb;
+        const double length = 1000; // [µm]
+        const double diam = 1; // [µm]
+
+        sample_tree samples({msample{0, 0, 0, 0.5*diam}, msample{length, 0, 0, 0.5*diam}}, {mnpos, 0u});
+        cable_cell c(samples);
+
+        c.paint(reg::all(), "hh"); // HH mechanism over whole cell.
+        c.place(mlocation{0, 0.}, i_clamp{0., INFINITY, 1.}); // Inject a 1 nA current indefinitely.
+        return c;
+    }
+};
+
+int main(int argc, char** argv) {
+    argv0 = argv[0];
+    try {
+        options opt;
+        if (!parse_options(opt, argc, argv)) {
+            return 0;
+        }
+
+        cable_recipe R(opt.probe_addr, opt.n_cv);
+
+        auto context = arb::make_context();
```

Comment:
Do we explicitly all the new probe features on GPU?

Can we add a flag to "turn on" running the sim on the GPU if one is available?

Reply:
I'm not sure I follow what you'd prefer here ... as things stand, the CPU ones are always tested, and the GPU versions are tested if there is one available and we've built GPU support in.

Reply:
Regarding the probe example, this comes back to the ongoing debate about the easy context stuff, #988.